proxy/reverse.go

```diff
@@ -134,6 +135,11 @@ func (p *reverseProxy) ServeHTTP(rw http.ResponseWriter, clientreq *http.Request
 		}
 		redirectRequest(proxyreq, ep.URL)
 
+		if repeated := p.checkProxyLoop(proxyreq, ep.URL.String()); repeated {
```

Comment:
@yichengq As far as I understand, when given multiple endpoints, we proxy a request only to one of the endpoints as in:

```go
for _, ep := range endpoints {
   ...
   res, err = p.transport.RoundTrip(proxyreq)
   ...
   break
}
```
So looks like this is where the looping is happening, and best place to detect if an endpoint
gets cycled with that transport. If we put this check to the beginning of the `ServeHTTP`, we
won't be able to know which endpoint gets the RoundTrip.

Or I might be wrong. Please correct me. Thanks!